### PR TITLE
improves ComprehensiveIT and adjust sunny ITs

### PR DESCRIPTION
**Pull request by keith-turner**
This change adds a new security test to Comprehensive IT and moves some test from ReadWriteIT into comprehensive IT.  Each test in ReadWriteIT spins up a mini cluster, which takes a lot of time for really simple test.  Also remove the sunny tag from shell ITs as ComprehensiveIT now covers most of what thost test were covering, but using public API instead of shell.

### CODE REVIEW

**Comment by keith-turner** [comment]
Running all of the sunny ITs with this changes looks like the following.

```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.accumulo.test.rpc.ThriftBehaviorIT
[INFO] Tests run: 12, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.544 s -- in org.apache.accumulo.test.rpc.ThriftBehaviorIT
[INFO] Running org.apache.accumulo.test.InMemoryMapIT
[INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.911 s -- in org.apache.accumulo.test.InMemoryMapIT
[INFO] Running org.apache.accumulo.test.functional.NativeMapIT
[INFO] Tests run: 12, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.737 s -- in org.apache.accumulo.test.functional.NativeMapIT
[INFO] Running org.apache.accumulo.test.functional.ReadWriteIT
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 65.15 s -- in org.apache.accumulo.test.functional.ReadWriteIT
[INFO] Running org.apache.accumulo.test.functional.WALSunnyDayIT
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 89.27 s -- in org.apache.accumulo.test.functional.WALSunnyDayIT
[INFO] Running org.apache.accumulo.test.ComprehensiveIT
[INFO] Tests run: 12, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 44.19 s -- in org.apache.accumulo.test.ComprehensiveIT
[INFO] Running org.apache.accumulo.test.start.KeywordStartIT
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.436 s -- in org.apache.accumulo.test.start.KeywordStartIT
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 52, Failures: 0, Errors: 0, Skipped: 0
```

File: test/src/main/java/org/apache/accumulo/test/ComprehensiveIT.java [hunk 1]
```diff
@@ -417,6 +426,299 @@ public void testOverwriteAndBatchDeleter() throws Exception {
...
+        assertEquals(splits, Set.copyOf(userClient.tableOperations().listSplits(rootsTable)));
+
+        // user should not have permission to bulk import
+        ase = assertThrows(AccumuloSecurityException.class, () -> bulkImport(userClient, rootsTable,
+            List.of(generateKeys(200, 250, tr -> true), generateKeys(250, 300, tr -> true))));
+        assertEquals(SecurityErrorCode.PERMISSION_DENIED, ase.getSecurityErrorCode());
+        verifyData(userClient, rootsTable, AUTHORIZATIONS, generateKeys(0, 200));
+
+        // TODO open a bug about this, had to add this permission to get bulk import to work
+        client.securityOperations().grantSystemPermission("user1", SystemPermission.SYSTEM);
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/ComprehensiveIT.java:610]: @ctubbsii  I had to grant the SYSTEM permission for bulk import to work.  Curious if you know anything about this.  Have not had a chance to look into it yet, wondering if its a bug.

File: test/src/main/java/org/apache/accumulo/test/functional/LocalityGroupIT.java [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+import org.apache.accumulo.core.metadata.schema.MetadataSchema;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.accumulo.harness.AccumuloClusterHarness;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.Text;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LocalityGroupIT extends AccumuloClusterHarness {
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/functional/LocalityGroupIT.java:62]: everything in this IT was copied exactly from ReadWriteIT, so no new code here.  Just wanted to move these test out of the sunny profile.

**Review by dlmarion — APPROVED** [review]

File: test/src/main/java/org/apache/accumulo/test/ComprehensiveIT.java (hunk 1 above)
Comment (ctubbsii) [reply test/src/main/java/org/apache/accumulo/test/ComprehensiveIT.java:610]: @keith-turner I think you could have granted just `TablePermission.BULK_IMPORT` for that specific user on that specific table. To bulk import, `WRITE` is insufficient. There's actually a `BULK_IMPORT` permission.

Comment (keith-turner) [reply test/src/main/java/org/apache/accumulo/test/ComprehensiveIT.java:610]: @ctubbsii a few lines further down BULK_IMPORT is granted, but the SYSTEM permission also has to be granted for it to work.

Comment (ctubbsii) [reply test/src/main/java/org/apache/accumulo/test/ComprehensiveIT.java:610]: That seems strange to me. Might be a bug. Did you ever create a follow-on issue about this?

**Pull request merged**